### PR TITLE
DType Serialization

### PR DESCRIPTION
**Pull request by gatesn**
* Clean up DType serde
* Create build-vortex package to unify flatbuffer/protobuffer code generation.
* Add protobuf DType definitions.

FLUP questions:
* Should we remove nullable from DType and define a separate `FieldType { dtype: DType, nullable: bool }`? Recursive types such as Struct and List would then use FieldType. It does introduce an interesting quirk of a nullable null... But maybe that's less annoying that duplicating nullability everywhere?
* If we don't want a separate FieldType, we could still pull `nullable` up in both flatbuffer and protobuf definitions.

Start of #277

### CODE REVIEW

File: vortex-dtype/proto/dtype.proto [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  repeated DType dtypes = 2;
+  bool nullable = 3;
+}
+
+message List {
+  DType element_type = 1;
+  bool nullable = 2;
+}
+
+message Extension {
```
Comment (gatesn) [inline vortex-dtype/proto/dtype.proto:55]: Do we want these messages nested inside the DType message?

Comment (robert3005) [reply vortex-dtype/proto/dtype.proto:55]: I don't get this comment - where else would you put it? Do you want to do union of DType and Extension?

Comment (gatesn) [reply vortex-dtype/proto/dtype.proto:55]:
```
message DType {
  message Primitive {..}
  message Extension {..}
}
```

It's purely a name spacing thing

Comment (robert3005) [reply vortex-dtype/proto/dtype.proto:55]: oh, that's what you mean. I haven't touched proto for too long so I forgot this is a thing you can do. My general rule was nest the messages unless there was some broader applicability.

Comment (gatesn) [reply vortex-dtype/proto/dtype.proto:55]: Actually I've decided against this _because_ there is broader applicability, e.g. a Struct dtype being used as a schema (if we pull up nullability, but that's a PR for later)

**Pull request merged**